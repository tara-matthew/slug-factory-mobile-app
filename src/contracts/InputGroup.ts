import {IInputWithLabelProps} from "./InputWithLabel";

export interface IInputGroupProps {
    inputs: IInputWithLabelProps[];
    sendDataToParent: (data: never) => void;
}
