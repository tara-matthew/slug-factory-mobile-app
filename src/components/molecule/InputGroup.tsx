import React from "react";
import { View } from "react-native";
import InputWithLabel from "./InputWithLabel";
import { IInputGroupProps } from "../../contracts/InputGroup";

// TODO Would it be possible to set a variant on the inputWithLabel, rather than spreading the inputProps? ie autocapitalised, secureText
// Could be over-engineered as a solution though
const InputGroup = ({ inputs, sendDataToParent }: IInputGroupProps) => {
    return (
        <View>
            {inputs.map((field, index) => (
                <View key={ index }>
                    <InputWithLabel
                        title={ field.title }
                        inputProps={ { ...field.inputProps } }
                        value={ field?.value }
                        defaultValue={ field?.defaultValue }
                        placeholder={ field.placeholder }
                        isMultiline={ field.isMultiline }
                        sendDataToParent={ sendDataToParent }
                    >

                    </InputWithLabel>
                </View>
            ))}
        </View>
    );
};

export default InputGroup;
