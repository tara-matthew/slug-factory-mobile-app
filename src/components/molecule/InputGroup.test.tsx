import React from "react";
import { render } from "@testing-library/react-native";
import InputGroup from "./InputGroup";

it("renders correctly", () => {
    const mockSendDataToParent = jest.fn();
    const inputs = [
        {
            title: "First input",
            placeholder: "First placeholder",
            isMultiline: false,
        },
        {
            title: "Second input",
            placeholder: "Second placeholder",
            isMultiline: false,
        },
        {
            title: "Third input",
            placeholder: "Third placeholder",
            isMultiline: false,
        },
    ];

    const {toJSON} = render(<InputGroup  inputs={inputs} sendDataToParent={mockSendDataToParent} />);
    expect(toJSON()).toMatchSnapshot();

});

it("displays all fields correctly", () => {
    const mockSendDataToParent = jest.fn();
    const inputs = [
        {
            title: "First input",
            placeholder: "First placeholder",
            isMultiline: false,
        },
        {
            title: "Second input",
            placeholder: "Second placeholder",
            isMultiline: false,
        },
        {
            title: "Third input",
            placeholder: "Third placeholder",
            isMultiline: false,
        },
    ];

    const {getByText, getByPlaceholderText} = render(<InputGroup  inputs={inputs} sendDataToParent={mockSendDataToParent} />);

    const placeholders = [getByPlaceholderText("First placeholder"), getByPlaceholderText("Second placeholder"), getByPlaceholderText("Third placeholder")];
    placeholders.forEach(placeholder => expect(placeholder).toBeTruthy());

    const inputTexts = [getByText("First input"), getByText("Second input"), getByText("Third input")];
    inputTexts.forEach(input => expect(input).toBeTruthy());
})
