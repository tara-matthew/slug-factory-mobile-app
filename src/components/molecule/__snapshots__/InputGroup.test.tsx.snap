// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View>
      <Text
        className="font-bold text-md"
      >
        First input
      </Text>
      <View
        className="bg-black/5 w-full p-5 rounded-2xl mb-7"
      >
        <TextInput
          defaultValue={null}
          onChangeText={[Function]}
          placeholder="First placeholder"
          value={null}
        />
      </View>
    </View>
  </View>
  <View>
    <View>
      <Text
        className="font-bold text-md"
      >
        Second input
      </Text>
      <View
        className="bg-black/5 w-full p-5 rounded-2xl mb-7"
      >
        <TextInput
          defaultValue={null}
          onChangeText={[Function]}
          placeholder="Second placeholder"
          value={null}
        />
      </View>
    </View>
  </View>
  <View>
    <View>
      <Text
        className="font-bold text-md"
      >
        Third input
      </Text>
      <View
        className="bg-black/5 w-full p-5 rounded-2xl mb-7"
      >
        <TextInput
          defaultValue={null}
          onChangeText={[Function]}
          placeholder="Third placeholder"
          value={null}
        />
      </View>
    </View>
  </View>
</View>
`;
